core/src/banking_stage.rs

```diff
@@ -1336,6 +1344,18 @@ impl BankingStage {
                         let post_balances = bank.collect_balances(batch);
                         let post_token_balances =
                             collect_token_balances(bank, batch, &mut mint_decimals);
+                        let mut transaction_indexes_iter = transaction_indexes.into_iter();
+                        let batch_transaction_indexes: Vec<_> = tx_results
+                            .execution_results
+                            .iter()
+                            .map(|result| {
+                                if result.was_executed() {
+                                    transaction_indexes_iter.next().unwrap_or_default()
+                                } else {
+                                    0
+                                }
+                            })
+                            .collect();
```

Comment:
The randomization in blockstore_processor is why we have to construct the `batch_transaction_indexes` here. The indexes need to be set and sent to the TransactionStatusService, otherwise that service can't re-engineer the randomization.